### DISCUSSION TITLE
How to use in .NET8 WPF application (or offscreen but .NET8)

### DISCUSSION THREAD

**Discussion by lofti198** [cefsharp/CefSharp#4787] [Q&A] [answered]
I tried to install packages to WPF .NET8 project (Cefsharp.Common, CefsharpOffscreen, Cefsharp.WPF), but it seems that there is some issue:
![image](https://github.com/cefsharp/CefSharp/assets/42960488/e2669274-89d2-4b00-8a83-56a22b74bce3)

One of errors is:
The type 'IRenderWebBrowser' is defined in an assembly that is not referenced. You must add a reference to assembly 'CefSharp, Version=123.0.60.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138'

What is actually "CefSharp" package? How to find it at nuget?

![image](https://github.com/cefsharp/CefSharp/assets/42960488/77efbe33-b74e-443a-bf91-69525859a694)

**Comment by amaitland** [comment, marked as answer]
> .NET8 project (Cefsharp.Common, CefsharpOffscreen, Cefsharp.WPF),

These packages are for `.Net Framework`. You need to use the `NetCore` packages.

https://github.com/cefsharp/CefSharp/wiki/Quick-Start-For-MS-.Net-5.0-or-greater

**Reply by lofti198** [reply]
Thank you! Working awesome!

**Reply by mwpowellhtx** [reply]
Question, is it good for screen scraping on or off screen? Automation?

Better still, if possible, I have a need to listen for [_out bound RESTful requests_](4856), if that were possible. Not necessarily page address changes, but internally there are RESTful things going on.

**Reply by lofti198** [reply]
> Question, is it good for screen scraping on or off screen? Automation?
> 
> Better still, if possible, I have a need to listen for [_out bound RESTful requests_](4856), if that were possible. Not necessarily page address changes, but internally there are RESTful things going on.

I guess I have very similar task. Do you mean - to intercept network traffic from the webpage (like AJAX requests etc)?

**Reply by mwpowellhtx** [reply]
Yes, not necessarily the page address, but RESTful requests happening besides that address. If it is AJAX, it may be, I do not know, but I can see the requests happening in my Chrome network traffic. I need to intercept those, or at least record the request address, or other diagnostics.

**Reply by mwpowellhtx** [reply]
To elaborate a little, I was able to initiate the sequence using a rather simple, naive `HttpClient` based approach, and it responds successfully, but that is as far as it gets. There is no doubt proper 1C browser behavior involved, renders the page, redirects to another address, etc, and then we see the rest of the sequence, RESTful requests, so on and so forth. So I wanted to consider a proper embedded, or even off screen, browser solution.

**Reply by lofti198** [reply]
I guess, what you are searching for you can implement with CustomResourceRequestHandler (I previously used it to intercept image loading by website, actually to avoid image loading).

**Reply by mwpowellhtx** [reply]
> I guess, what you are searching for you can implement with CustomResourceRequestHandler (I previously used it to intercept image loading by website, actually to avoid image loading).

Perhaps, maybe. I would need to investigate that a bit further. I'm not imposing any prohibitions on loading anything, per se, in fact, it needs to send the request and receive response in as unimpeded a manner as possible. I literally just need to parse the request Url, arguments, etc. And get out of the way quickly as possible. If that makes sense. Thanks!
